internal/migrate/versions.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package migrate
+
+import (
+	"database/sql"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+)
+
+/*
+
+Scenarios it doesn't handle properly.
+
+1. Rollback to initial state on multi-step migration.
+
+	Let's say there's a scenario where we run migration steps:
+	1. update a table schema
+	2. move files
+	3. update a table schema
+	4. update a table schema, which fails
+
+	In this case there's no easy way to rollback the moving of files.
+
+2. Undoing migrations.
+
+	Intentionally left out, because we do not gain that much from currently.
+
+3. Snapshotting the whole state.
+
+	This probably should be done by the user of this library, when there's disk-space available.
+
+4. Figuring out what the exact executed steps are.
+*/
+
+// Migration describes a migration steps
+type Migration struct {
+	Table    string
+	OnCreate Action
```

Comment:
Is OnCreate substantially different than Steps[0]?  Should we simplify this more?

Reply:
Yes, @mniewrzal will remove it.

Reply:
its already removed ;) just refresh github